ambry-router/src/main/java/com/github/ambry/router/UndeleteOperation.java

```diff
@@ -95,8 +98,20 @@ public class UndeleteOperation {
     this.callback = callback;
     this.time = time;
     this.operationTimeMs = operationTimeMs;
-    this.operationTracker = new UndeleteOperationTracker(routerConfig, blobId.getPartition(),
-        clusterMap.getDatacenterName(blobId.getDatacenterId()), routerMetrics);
+    if (routerConfig.routerUndeleteOperationTrackerType.equals(UndeleteOperationTracker.class.getSimpleName())) {
```

Comment:
nit: consider using enum.

Reply:
the reason not to use enum here is that we have same logic for GetBlobOperation and PutBlobOperation, and they are not using enum either. Want to keep it consistent.